packages/aws-cdk-lib/aws-appsync/lib/eventapi.ts

```diff
@@ -0,0 +1,591 @@
+    const logsRoleArn: string =
+      config.role?.roleArn ??
+      new Role(this, 'ApiLogsRole', {
+        assumedBy: new ServicePrincipal('appsync.amazonaws.com'),
+        managedPolicies: [ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSAppSyncPushToCloudWatchLogs')],
+      }).roleArn;
+    const fieldLogLevel: FieldLogLevel = config.fieldLogLevel ?? FieldLogLevel.NONE;
+    return {
+      cloudWatchLogsRoleArn: logsRoleArn,
+      logLevel: fieldLogLevel,
+    };
+  }
+
+  private setupAuthProviderTypes(authProviders?: AuthorizationMode[]) {
+    if (!authProviders || authProviders.length === 0) return [AuthorizationType.API_KEY];
+    const modes = authProviders.map((mode) => mode.authorizationType);
+    return modes;
+  }
+
+  private mapAuthorizationProviders(authProviders: AuthorizationMode[]) {
+    return authProviders.reduce<CfnApi.AuthProviderProperty[]>((acc, mode) => {
+      acc.push({
+        authType: mode.authorizationType,
+        cognitoConfig: setupCognitoConfig(mode.cognitoConfig),
+        openIdConnectConfig: setupOpenIdConnectConfig(mode.openIdConnectConfig),
+        lambdaAuthorizerConfig: setupLambdaAuthorizerConfig(mode.lambdaAuthorizerConfig),
+      });
+      return acc;
+    }, []);
+  }
+
+  private mapAuthorizationConfig(authModes: AuthorizationType[]) {
+    return authModes.map((mode) => ({ authType: mode }));
+  }
+
+  private validateAuthorizationProps(authProviders: AuthorizationMode[]) {
+    const keyConfigs = authProviders.filter((mode) => mode.authorizationType === AuthorizationType.API_KEY);
+    const someWithNoNames = keyConfigs.some((config) => !config.apiKeyConfig?.name);
+    if (keyConfigs.length > 1 && someWithNoNames) {
+      throw new Error('You must specify key names when configuring more than 1 API key.');
+    }
+
+    if (authProviders.filter((authProvider) => authProvider.authorizationType === AuthorizationType.LAMBDA).length > 1) {
+      throw new Error(
+        'You can only have a single AWS Lambda function configured to authorize your API. See https://docs.aws.amazon.com/appsync/latest/devguide/security.html',
+      );
+    }
+
+    if (authProviders.filter((authProvider) => authProvider.authorizationType === AuthorizationType.IAM).length > 1) {
+      throw new Error("You can't duplicate IAM configuration. See https://docs.aws.amazon.com/appsync/latest/devguide/security.html");
+    }
+
+    authProviders.map((authProvider) => {
+      if (authProvider.authorizationType === AuthorizationType.OIDC && !authProvider.openIdConnectConfig) {
+        throw new Error('Missing OIDC Configuration');
+      }
+      if (authProvider.authorizationType === AuthorizationType.USER_POOL && !authProvider.cognitoConfig) {
+        throw new Error('Missing Cognito Configuration');
+      }
+      if (authProvider.authorizationType === AuthorizationType.LAMBDA && !authProvider.lambdaAuthorizerConfig) {
+        throw new Error('Missing Lambda Configuration');
```

Comment:
nit: seems too generic. we could add like
`Lambda authorization is specified but Lambda Authorizer Configuration is missing in the AuthProvider`

and similary for the other types